ARCs/arc-draft_alternative_0014.md

```diff
@@ -0,0 +1,285 @@
+- `nonce` representing a unique/random value generated by the verifier.
+
+The JSON structure for such an object is:
+
+```typescript
+interface AuthMessage {
+ /** The domain name of the service */
+ service: string;
+ /** Optional, description of the service */
+ desc?: string;
+ /** Algorand account to authenticate with*/
+ authAcc: string;
+ /** Challenge generated server-side */
+ nonce: string;
+}
+```
+
+For example:
+
+```json
+{
+ "service": "www.servicedomain.com",
+ "desc": "Domain offers important services to users",
+ "authAcc": "KTGP47G64KCXWJS64W7SGJNKTHE37TYDCI64USXI3XOYE6ZSH4LCI7NIDA",
+ "nonce": "1234abcde!%",
+}
+```
+
+The `nonce` field **MUST** be unique for each authentication and **MUST NOT** be used more than once to avoid replay attacks.
+
+Users sign *Authentication Message*s using the secret keys stored into a wallet.
+
+***Problem***: Most of the Algorand wallets can only sign Algorand `Transaction` objects, and do not offer the possibility to sign arbitrary bytes. To overcome such a limitation, the ARC-0014 distinguishes two authentication objects, namely the **Simple Authentication Message**, and the **Transaction Authentication Message**. The former **SHOULD** be used by any signing mechanisms able to sign random bytes, whereas the latter can be used with almost any Algorand wallet.
+
+#### Simple Authentication Message
+
+The *Simple Authentication Message* is a sequence of bytes representing an ARC-0014 *Authentication Message*. It **SHOULD** be used in contexts that support signatures of random bytes. A *Simple Authentication Message* is an *Authentication Message* prepended with the prefix `arc14` for domain separation. It **MUST** be represented as the hash SHA-512/256 of such prefix, together with a msgpack encoded *Authentication Message*. For example, given an *Authentication Message* object `aut_message`, its *Simple Authentication Message* representation would be something like: `SHA512_256(“arc14”+msgpacked_auth_message)`.
+
+#### Transaction Authentication Message
+
+A *Transaction Authentication Message* is an *Authentication Message* represented as an Algorand `Transaction` object. Such a transaction must have the following characteristics:
+
+- it **SHOULD** be an Algorand `Payment Transaction`;
+- it **MUST** include a *Simple Authentication Message* into the transaction's `note` field;
+- it **MUST** be *invalidated* i.e. not executable on any official Algorand network.
+
+> Malicious users could intercept and execute it causing unexpected consequences. For instance, in case of a **MainNet** transaction, it could be intercepted by a malicious user and executed, burning some Algos from the sender’s account due to the payment of fees.
+
+The fields of a `Payment Transaction` object which represents a *Transaction Authentication Message* **MUST** be initialized as follows
+
+- `amount` = 0;
+- `sender/receiver` = *PKa*;
+- `firstValid/lastValid` = 0;
+- `fee` = 0;
+- `genesisId` = “arc14-arc”;
```

Comment:
I have noticed that if the genesis ID in the authentification transaction does not match that of the one set in either walletConnect or Algosigner, these applications will refuse allow the user to sign the authentification message. Therefore it seems like a necessity to create a custom "arc14-arc" network for a user's algosigner extension or for walletconnect, have their wallet application create this custom network. Adds complexity to the end user experience.  

In addition, when signing a transaction with the first round and last round equal to zero, algosigner and walletconnect will refuse to enable the user to sign the transaction as well, I have found this to be the case on the test network.

Reply:
A zero fee works

Reply:
Thanks for your comments @multivac88. Whats the configuration you used to make it work? If genesis id/hash and firstvalid/lastvalid are not working, I think we are fine with a self-payment txns with 0 fee to make it invalid. @barnjamin/@cusma WDYT?

Reply:
payment to self with 0 amount and 0 fee and null group id should be good, but id also like to see if first/last valid of 1 might work? can you try it?